mmcv/runner/base_runner.py

```diff
@@ -360,17 +369,25 @@ def resume(self,
             checkpoint = self.load_checkpoint(
                 checkpoint, map_location=map_location)
 
+        # Note that runner may resume a checkpoint in the middle of workflows
+        # or epoch and resuming checkpoint will reset workflow and dataloader
         self._epoch = checkpoint['meta']['epoch']
         self._iter = checkpoint['meta']['iter']
         if self.meta is None:
             self.meta = {}
-        self.meta.setdefault('hook_msgs', {})
-        # load `last_ckpt`, `best_score`, `best_ckpt`, etc. for hook messages
-        self.meta['hook_msgs'].update(checkpoint['meta'].get('hook_msgs', {}))
-
-        # Re-calculate the number of iterations when resuming
-        # models with different number of GPUs
-        if 'config' in checkpoint['meta']:
+        else:
+            # use current env information and seed if available
+            # instead of resuming them from checkpoint
+            if self.meta.get('env_info', None) is not None:
+                checkpoint['meta'].update(env_info=self.meta['env_info'])
```

Comment:
Why we need to update the `env_info` for `checkpoint['meta']`?

Reply:
The `env_info` such as `device` or `library` might be changed.

Reply:
The whole `checkpoint['meta']` will be updated to `self.meta`